### PR TITLE
Allow viewing in-band metadata tracks from shimmed texttracklist

### LINKED ISSUE #2266: In-band metadata tracks should be visible when emulated text tracks are active
If javascript-emulated text tracks are being used, `player.textTracks()` should return at least all the tracks that are returned from `video.textTracks`.

## Steps to Reproduce
1. Create a player and force emulated text tracks in Safari
1. Load a stream with an embedded metadata track
1. Call `player.textTracks()`

The returned TextTrackList should include a metadata text track but it will be empty.

### PR DESCRIPTION
**Pull request by gkatsev**
Fixed #2266.

### CODE REVIEW

File: src/js/media/html5.js [hunk 1]
```diff
@@ -61,6 +61,9 @@ vjs.Html5 = vjs.MediaTechController.extend({
 
     if (this['featuresNativeTextTracks']) {
       this.on('loadstart', vjs.bind(this, this.hideCaptions));
+    } else {
+      // allow us to view the in-band metadata tracks from our shimmed TextTrackList
+      this.on('loadedmetadata', vjs.bind(this, this.addInBandMetadataTracks));
```
Comment (dmlap) [inline src/js/media/html5.js]: I'm not sure that inband tracks will show up by `loadedmetadata`. Did you find this in the spec or discover it experimentally?

Comment (gkatsev) [inline src/js/media/html5.js]: Experimentally.

Comment (gkatsev) [inline src/js/media/html5.js]: Though, this reminds me. I probably need to remove the old ones from the list on source change.

Comment (dmlap) [inline src/js/media/html5.js]: An in-band metadata track might not be evident until a decent amount of buffering has occurred. If there were a way to make the track registration a bit more robust for that case, that would be great.

Comment (gkatsev) [inline src/js/media/html5.js]: I'll take a look whether the `addtrack` event fires for it. Might listen to both or something.

**Comment by gkatsev** [comment]
This should handle multiple in-band metadata tracks better and should be able to clear them out properly between video loads.
I'm not sure how easy it would be to test this.
Also, this would also need to be ported to master as well.

File: src/js/media/html5.js [hunk 2]
```diff
@@ -184,6 +193,39 @@ vjs.Html5.prototype.hideCaptions = function() {
   }
 };
 
+vjs.Html5.prototype.addInBandMetadataTracks = function(event) {
+  var tracks = this.textTracks(),
+      newTrack = event.track || null,
+      track,
+      i = 0;
+
+  for (i = 0; i < tracks.length; i++) {
```
Comment (dmlap) [inline src/js/media/html5.js]: This shouldn't be necessary unless `addtrack` fired twice, right?

**Comment by dmlap** [comment]
We should have a test for this.

File: src/js/media/html5.js (hunk 2 above)
Comment (gkatsev) [inline src/js/media/html5.js]: Yeah, I'm not certain it'll be necessarily useful. Even if you re-apply the same video you get a new track.
This was kind of left-over from when I had the removal of tracks in the same method as additions and so far I left it for "extra security" but I'm not certain it actually buys us much.

**Comment by gkatsev** [comment]
I'll try and come up with some tests.

**Comment by gkatsev** [comment]
This will become obsolete as part of the work to fix #2367. I'm using this general framework for that fix anyway. Given that we've decided not to include it in stable, I'm going to close this.

**Pull request closed without merging**